url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/request/HttpGetRequester.java

```diff
@@ -0,0 +1,141 @@
+package net.hetimatan.net.http.request;
+
+
+import java.io.IOException;
+
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.event.net.MessageSendTask;
+import net.hetimatan.util.event.net.io.KyoroSelector;
+import net.hetimatan.util.event.net.io.KyoroSocket;
+import net.hetimatan.util.event.net.io.KyoroSocketImpl;
+import net.hetimatan.util.http.HttpObject;
+import net.hetimatan.util.http.HttpRequest;
+
+
+public class HttpGetRequester  {
+	public static final String REQUEST_METHOD_GET = "GET";
+	public static final String SCHEME_HTTP = "http";
+	public static final String HTTP10 = "HTTP/1.0";
+	public static final String HTTP11 = "HTTP/1.1";
+	public static final String HEADER_HOST = "Host";
+	public static final String USER_AGENT = "User-Agent";
+	public static final String CONTENT_LENGTH = "Content-Length";
+
+	private HttpRequestBuilder mBuilder = new HttpRequestBuilder();
+	private MessageSendTask mTask = null;
+	private CashKyoroFile mSendCash = null;
+
+	// --- <omake> ---
+	private boolean mIsPostMode = false;
+	public void setBody(KyoroFile body) {
+		mBuilder.putBody(body);
+	}
+	public void isPostMode(boolean on) {
+		mIsPostMode = on;
+	}
+	public boolean isPostMode() {
+		return mIsPostMode;
+	}
+	// --- </omake> ---
+
+	public HttpGetRequester() {
+	}
+
+	public static void log(String message) {
+		System.out.println("KyoroSocketGetRequester#"  + message);
+	}
+
+	public HttpRequestBuilder getUrlBuilder() {
+		return mBuilder;
+	}
+
+	public KyoroSocket connect(KyoroSocket socket) throws IOException, InterruptedException {
+		if(socket == null) {
+			socket = new KyoroSocketImpl();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *socket*. The resource is referenced at lines 54, 57, 58. The resource is returned at line 59. There are other execution paths that don&#39;t close the resource or return it.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).